packages/flutter_tools/lib/src/commands/build_macos.dart

```diff
@@ -21,11 +21,21 @@ class BuildMacosCommand extends BuildSubCommand {
     addCommonDesktopBuildOptions(verboseHelp: verboseHelp);
     usesFlavorOption();
     argParser
-      .addFlag('config-only',
+      ..addFlag('config-only',
         help: 'Update the project configuration without performing a build. '
           'This can be used in CI/CD process that create an archive to avoid '
           'performing duplicate work.'
-    );
+      )
+      ..addOption('darwin-arch',
```

Comment:
I'm reluctant to add a new flag to this command for such a very niche use-case https://github.com/flutter/flutter/issues/152221#issuecomment-2346862463